### PR TITLE
NXP-29020 implement table of contents on explorer pages

### PR DESCRIPTION
**Pull request by atchertchian**
T&P running at https://qa.nuxeo.org/jenkins/job/TestAndPush/job/ondemand-testandpush-atchertchian-master-2/87/

### CODE REVIEW

**Comment by atchertchian** [comment]
T&P passed ok

File: modules/platform/nuxeo-apidoc-server/nuxeo-apidoc-webengine/src/main/resources/skin/resources/script/jquery.toc.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+(function ($) {
+
+  $.fn.toc = function (tocListContainer, tocList) {
+    $(tocList).empty();
+
+    var index = 0;
```
Comment (akervern) [inline modules/platform/nuxeo-apidoc-server/nuxeo-apidoc-webengine/src/main/resources/skin/resources/script/jquery.toc.js:6]: you can now use proper variable scoping with `let` or `const`

File: modules/platform/nuxeo-apidoc-server/nuxeo-apidoc-webengine/src/main/resources/skin/resources/script/jquery.toc.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    $("h2").each(function () {
+      var name = "toc" + index;
+      var anchor = "<a name='" + name + "' class='anchor'></a>";
```
Comment (akervern) [inline modules/platform/nuxeo-apidoc-server/nuxeo-apidoc-webengine/src/main/resources/skin/resources/script/jquery.toc.js:9]: Use String templating for that: https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/template_strings

File: modules/platform/nuxeo-apidoc-server/nuxeo-apidoc-webengine/src/main/resources/skin/resources/script/jquery.toc.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+(function ($) {
+
+  $.fn.toc = function (tocListContainer, tocList) {
+    $(tocList).empty();
+
+    var index = 0;
+    $("h2").each(function () {
```
Comment (akervern) [inline modules/platform/nuxeo-apidoc-server/nuxeo-apidoc-webengine/src/main/resources/skin/resources/script/jquery.toc.js:7]: Apparently `jquery#each` callback already contains index and obj ref: https://api.jquery.com/jQuery.each/#jQuery-each-object-callback

File: modules/platform/nuxeo-apidoc-server/nuxeo-apidoc-webengine/src/main/resources/skin/resources/script/jquery.toc.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+      var li = "<li><a href='#" + name + "'>" + $(this).text() + "</a></li>";
+      $(tocList).append(li);
+      index++;
+    });
+
+    if (index < 2) {
+      // only one item ==> remove TOC
+      console.log("removing");
+      $(tocListContainer).remove;
```
Comment (akervern) [inline modules/platform/nuxeo-apidoc-server/nuxeo-apidoc-webengine/src/main/resources/skin/resources/script/jquery.toc.js:20]: It looks like you miss parenthesis here.

File: modules/platform/nuxeo-apidoc-server/nuxeo-apidoc-webengine/src/main/resources/skin/resources/script/jquery.toc.js (hunk 3 above)
Comment (atchertchian) [reply modules/platform/nuxeo-apidoc-server/nuxeo-apidoc-webengine/src/main/resources/skin/resources/script/jquery.toc.js:7]: even if i need the index outside of the iteration?

File: modules/platform/nuxeo-apidoc-server/nuxeo-apidoc-webengine/src/main/resources/skin/resources/script/jquery.toc.js (hunk 2 above)
Comment (atchertchian) [reply modules/platform/nuxeo-apidoc-server/nuxeo-apidoc-webengine/src/main/resources/skin/resources/script/jquery.toc.js:9]: i changed the logics, you'll tell me if you prefer or not

**Comment by atchertchian** [comment]
Relaunched https://qa.nuxeo.org/jenkins/job/TestAndPush/job/ondemand-testandpush-atchertchian-master-2/88/ after last changes

File: modules/platform/nuxeo-apidoc-server/nuxeo-apidoc-webengine/src/main/resources/skin/resources/script/jquery.toc.js (hunk 3 above)
Comment (akervern) [reply modules/platform/nuxeo-apidoc-server/nuxeo-apidoc-webengine/src/main/resources/skin/resources/script/jquery.toc.js:7]: you mean the list size?

Comment (atchertchian) [reply modules/platform/nuxeo-apidoc-server/nuxeo-apidoc-webengine/src/main/resources/skin/resources/script/jquery.toc.js:7]: i mean the number of toc items that were processed

Comment (atchertchian) [reply modules/platform/nuxeo-apidoc-server/nuxeo-apidoc-webengine/src/main/resources/skin/resources/script/jquery.toc.js:7]: i admit i originally copy/pasted the whole code, so i can rename the variable if it makes things clearer

Comment (atchertchian) [reply modules/platform/nuxeo-apidoc-server/nuxeo-apidoc-webengine/src/main/resources/skin/resources/script/jquery.toc.js]: Removed the use case after all (always show the TOC even if empty)

**Comment by atchertchian** [comment]
https://qa.nuxeo.org/jenkins/job/TestAndPush/job/ondemand-testandpush-atchertchian-master-2/89/ :heavy_check_mark:

**Pull request merged**